lib/matplotlib/figure.py

```diff
@@ -157,6 +158,59 @@ def __contains__(self, a):
         return a in self.as_list()
 
 
+class _AxesStack(object):
+    """Lightweight stack that tracks Axes in a Figure.
+    """
+
+    def __init__(self):
+        # We maintain a list of (creation_index, key, axes) tuples.
+        self._items = []
+        self._created = 0
+
+    def as_list(self):
+        """Copy of the list of axes, in the order of insertion.
+        """
+        return [ax for _, _, ax in sorted(self._items)]
```

Comment:
I know it's originally sorted this way, but I wonder if that's even necessary...